GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/GpsDataCaptureService.java

```diff
@@ -46,12 +43,13 @@
     private LocationManagerListener locationManagerListener;
     private FusedLocationProviderListener fusedLocationProviderListener;
 
-    private static File gpxDataFolder;
+    private static File gpxFileFolder;
     private static File gpxFile;
 
     private static TextView gpsDataTextView;
```

Comment:
Passing text veiew from activity to service is simple solution but introduces significant long term issues: exposes details of activity implementation to service. Could see MVC pattern as a solution. In general, will need to create interface that abstracts how activity presents locations and status updates. Then activity implements interface by setting text views as needed. When attaching to service will pass interface implementation instead of text views.

Reply:
You are right.  Maybe should adopt the Data Binding Library or LiveData? Which one would you suggest?

Reply:
You can use LiveData or you could create a simple callback as explained in     https://stackoverflow.com/questions/2463175/how-to-have-android-service-communicate-with-activity

Reply:
Experimented with MVVM pattern with LiveData, works.